### DISCUSSION TITLE
Manually refreshing Shared from the outside, eg from SwiftUI's `.refreshable { }`

### DISCUSSION THREAD

**Discussion by kamcma** [pointfreeco/swift-sharing#6] [Q&A] [answered]
During the beta of the shared tools in TCA, [an isowords PR demonstrated](https://github.com/pointfreeco/isowords/blob/b6e5a89e501adfb5629dc7090cab80da7ed7072d/Sources/AppFeature/AppView.swift#L241) a way to initiate a shared state refresh manually from the outside. This could be quite useful for state that is conceptually an external source of truth, but is not inherently observable and instead must be proactively fetched. For example: a response from a restful HTTP GET request, that is displayed in the SwiftUI `List` that is `.refreshable { ... }`.

At one point when I explored implementing the technique from the isowords PR myself, I was blocked by the fact that this appeared to rely on `SharedReader` [exposing the underlying persistence mechanism as a generic](https://github.com/pointfreeco/swift-composable-architecture/blob/4ce58fb2274c60ada9b21e8542d6ab86764f3782/Sources/ComposableArchitecture/SharedState/SharedReader.swift#L85-L89) that ended up not shipping in the final form of the tools.

Have there been any other explorations along these lines? I haven't had time to study all the new case studies in detail yet; do any of them have anything like this? Has anyone else explored what's possible with the public API as we know it from TCA?

**Comment by stephencelis** [comment, marked as answer]
@kamcma In this library's release we've introduced both `load` and `save` methods on `Shared` and `SharedReader` for eagerly requesting that any underlying shared persistence key does its thing. Does this help?

**Reply by kamcma** [reply]
Oh, that does sound like it could be exactly what I'm after. Apologies I didn't notice the new API; I assumed it would come over from TCA more or less unchanged. I'll mark this as answered and explore, and come back if I have more questions after. Thanks! Congrats on the launch of another library.
